src/transforms/scaling.jl

```diff
@@ -0,0 +1,54 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    ScaleCoords(sx, sy, sz)
+
+Transform geometry or mesh by scaling each coordinate of the vertices by 
+the corresponding factor. 
+
+## Arguments
+
+- `sx`: scaling factor for the first coordinate
+- `sy`: scaling factor for the second coordinate
+- `sz`: scaling factor for the third coordinate
+
+## Examples
+
+```julia
+ScaleCoords(1, 2, 3)
+```
+"""
+struct ScaleCoords{T<:Real} <: GeometricTransform
+  factors::AbstractVector{T}
```

Comment:
```suggestion
struct ScaleCoords{T} <: GeometricTransform
  factors::Vector{T}
```
AbstractVector as a field is not good for performance.